src/Compilers/Shared/CoreClrBuildClient.cs

```diff
@@ -46,14 +46,22 @@ protected override string GetSessionKey(BuildPaths buildPaths)
             return string.Empty;
         }
 
-        protected override async Task<BuildResponse> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string pipeName, string keepAlive, string libDirectory, CancellationToken cancellationToken)
+        protected override async Task<RunServerResult> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string pipeName, string keepAlive, string libDirectory, CancellationToken cancellationToken)
```

Comment:
Agree.  But the caller uses `Task.Result` which blocks until the value completes.  

Is the wording of the comment in the other file what's bothering you here?